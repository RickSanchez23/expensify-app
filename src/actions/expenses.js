import uuid from "uuid";

export const addExpense = ({
  amount = 0,
  description = "",
  note = "",
  createAt = 0,
  id = uuid(),
} = {}) => ({
  type: "ADD_EXPENSE",
  expense: {
    id,
    amount,
    description,
    note,
    createAt,
  },
});

export const deleteExpense = (itemId) => ({
  type: "REMOVE_EXPENSE",
  itemId,
});

export const editExpense = (itemId, updates) => ({
  type: "EDIT_EXPENSE",
  itemId,
  updates,
});
