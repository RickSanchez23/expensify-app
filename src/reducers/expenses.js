const expenseReducerDefaultState = [];
export default (state = expenseReducerDefaultState, action) => {
  switch (action.type) {
    case "ADD_EXPENSE":
      return [...state, action.expense];

    case "REMOVE_EXPENSE":
      return state.filter((item) => item.id !== action.itemId);

    case "EDIT_EXPENSE":
      return state.map((item) => {
        if (item.id === action.itemId) {
          return { ...item, ...action.updates };
        } else {
          return item;
        }
      });

    case "SET_EXPENSES" :
      return action.expenses

    default:
      return state;
  }
};
