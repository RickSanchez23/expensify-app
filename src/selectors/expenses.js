export const getVisibleExpenses = (
  expenses,
  { text, startDate, endDate, sortBy }
) => {
  return expenses
    .filter((item) => {
      const textFilterMatch = item.description
        .toLowerCase()
        .includes(text.toString());
      const startDateMatch =
        typeof startDate !== "number" || item.createAt >= startDate;
      const endDateMatch =
        typeof endDate !== "number" || item.createAt <= endDate;
      return textFilterMatch && startDateMatch && endDateMatch;
    })
    .sort((a, b) => {
      if (sortBy === "date") {
        return a.createAt < b.createAt ? 1 : -1;
      } else {
        return a.amount < b.amount ? 1 : -1;
      }
    });
};
