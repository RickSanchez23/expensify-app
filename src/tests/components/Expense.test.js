import React from "react";
import { shallow } from "enzyme";
import Expenses from "../../component/Expense";
import expenses from "../fixtures/expenses";
import moment from "moment";

test('should render Expense list item' , () => {
    const wrapper = shallow(<Expenses {...expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
})
