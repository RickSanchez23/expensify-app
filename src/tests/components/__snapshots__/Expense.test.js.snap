// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense list item 1`] = `
<div>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/edit/1"
  >
    <h3>
      Rent
    </h3>
  </NavLink>
  <p>
    310
    $ - 
    -345600000
  </p>
</div>
`;
